app/routes/login-callback/route.tsx

```diff
@@ -0,0 +1,37 @@
+import { LoaderFunctionArgs, redirect } from '@remix-run/node';
+import { commitSession, getSession, initializeEcomApiForRequest } from '~/lib/ecom/session';
+
+export async function loader({ request }: LoaderFunctionArgs) {
+    const session = await getSession(request.headers.get('Cookie'));
+    const storedOauthData = session.get('oAuthData');
+
+    if (storedOauthData === undefined) {
+        return redirect('/login');
+    }
+
+    const api = await initializeEcomApiForRequest(request);
+    const wixClient = api.getWixClient();
+
+    const returnedOAuthData = wixClient.auth.parseFromUrl(request.url, 'query');
+    if (returnedOAuthData.error) {
+        throw new Error(`Error: ${returnedOAuthData.errorDescription}`);
+    }
+
+    const memberTokens = await wixClient.auth.getMemberTokens(
+        returnedOAuthData.code,
+        returnedOAuthData.state,
+        storedOauthData,
+    );
+    wixClient.auth.setTokens(memberTokens);
+
+    session.set('wixEcomTokens', memberTokens);
+
+    const redirectUrl = new URL(storedOauthData.originalUri || new URL(request.url).host);
+    redirectUrl.searchParams.set('login', 'success');
```

Comment:
> redirectUrl.searchParams.set('login', 'success');

What is this for?

Reply:
this is done because netlify intentionally left original query parameters on redirects.
the flow:
1. user goes to `/login` - login process is initiated and user is redirected to wix login page
2. user logs in on wix login page
3. wix login page redirects user to `/login-callback?code=<oAuth login code>&state=<oAuth login state>`
4. our loader finishes login process and redirect user to original page (from where login was initiated) with 302 response code and locaiton like `/categories/all-products`
5. netlify keeps query params and user is redirected to `/categories/all-products?code=<oAuth login code>&state=<oAuth login state>`

in order to avoid `code` and `state` presence in URL there are two options:
1. `useEffect` and clear search params
2. add own search param when redirecting

we've discussed this with @yurii-ve  and decided it is better to add dummy `login=success` parameter when redirecting instead of clearing `code` and `state` params.
also, we haven't added a comment about it in code, cause we wanted to omit any netlify mentions, as it is not directly related to a template itself

link to netlify support answer that this is expected and done intentionally:
https://answers.netlify.com/t/netlify-keeps-searchparams-when-redirecting-in-my-next-js-app/108590/2

Reply:
Things like this absolutely must have a comment in code. This code looks unused and I could have easily deleted it during a refactor.

Reply:
> cause we wanted to omit any netlify mentions

We won't explain what this code is for, because we don't talk about Ne*lify around here.

Reply:
@olehrakwix sorry, probably removing this comment wasn't among the best of my ideas 😅

Reply:
![image](https://github.com/user-attachments/assets/0137dc4c-2fad-413d-a1ac-71b8bf14b386)

Reply:
added comment